analyses/tp53_nf1_score/run_classifier-tumor_purity_threshold.sh

```diff
@@ -0,0 +1,82 @@
+
+# We want to skip the poly-A steps in CI
+# if POLYA=1, poly-A steps will be run
+POLYA=${OPENPBTA_POLYAPLOT:-1}
+
+
+# This script should always run as if it were being called from
+# the directory it lives in.
+analysis_dir="$(perl -e 'use File::Basename;
+  use Cwd "abs_path";
+  print dirname(abs_path(@ARGV[0]));' -- "$0")"
+cd "$analysis_dir" || exit
+
+data_dir="../../data"
+scratch_dir="../../scratch/tp53-classifier"
+# Make sure scratch directory exists
+mkdir -p $scratch_dir
+
+# cds gencode bed file
+cds_file="${scratch_dir}/gencode.v27.primary_assembly.annotation.bed"
+snvconsensus_file="${data_dir}/pbta-snv-consensus-mutation.maf.tsv.gz"
+cnvconsensus_file="${data_dir}/consensus_seg_annotated_cn_autosomes.tsv.gz"
+
+histology_file="../../data/pbta-histologies.tsv"
+
+
+# Step 1: Prep the SNV consensus data for evaluation downstream
+# This only needs to be run if the output file `TP53_NF1_snv_alteration.tsv` does not already exist.
+if [ ! -f  ${analysis_dir}/results/TP53_NF1_snv_alteration.tsv ]; then
+  # Convert GTF to BED file
+  # Here we are only extracting lines with as a CDS i.e. are coded in protein
+  gunzip -c ${data_dir}/gencode.v27.primary_assembly.annotation.gtf.gz \
+    | awk '$3 ~ /CDS/' \
+    | convert2bed --do-not-sort --input=gtf - \
+    > $cds_file
+
+  # Prep the SNV consensus data for evaluation downstream
+  Rscript --vanilla ${analysis_dir}/00-tp53-nf1-alterations.R \
+    --snvConsensus ${snvconsensus_file} \
+    --cnvConsensus ${cnvconsensus_file} \
+    --histologyFile ${histology_file} \
+    --outputFolder ${analysis_dir}/results \
+    --gencode ${cds_file}
+fi
+
+# expression files for prediction from data release
+collapsed_stranded="${data_dir}/pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds"
+
+# Run classifier and ROC plotting for stranded data, with tumor purity threshold turned on (`-t` flag)
+python3 ${analysis_dir}/01-apply-classifier.py -f ${collapsed_stranded} -t
+
+# Run notebook to produce the `loss_overlap_domains_tp53` file consumed by 05-tp53-altered-annotation.Rmd
+Rscript -e "rmarkdown::render('${analysis_dir}/03-tp53-cnv-loss-domain.Rmd',params=list(tumor_purity_threshold = 1))"
+
+# Run notebook to produce the `fusion_bk_tp53_loss` and `sv_overlap_tp53` files consumed by 05-tp53-altered-annotation.Rmd
+Rscript -e "rmarkdown::render('${analysis_dir}/04-tp53-sv-loss.Rmd',params=list(tumor_purity_threshold = 1))"
+
+# Run notebook to produce `tp53_altered_status` file consumed by 06-evaluate-classifier.py
+Rscript -e "rmarkdown::render('${analysis_dir}/05-tp53-altered-annotation.Rmd',params=list(tumor_purity_threshold = 1))"
+
+# Produce files needed for ROC
+python3 ${analysis_dir}/06-evaluate-classifier.py \
+  -s ${analysis_dir}/results/tp53_altered_status_tumor-purity-threshold.tsv \
+  -f ${analysis_dir}/results/pbta-gene-expression-rsem-fpkm-collapsed.stranded_classifier_scores_tumor-purity-threshold.tsv \
+  -c ${histology_file} \
+  -o stranded_tumor-purity-threshold
```

Comment:
Is this missing `--apply_threshold`?

Reply:
Yes, but now that flag is going away: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1664/files#r1107354897